Client/Frontend/Browser/TabManager.swift

```diff
@@ -294,13 +308,10 @@ extension TabManager {
         var tabToSelect: Browser?
         if let savedTabs = coder.decodeObjectForKey("tabs") as? [SavedTab] {
             for savedTab in savedTabs {
-                if let url = savedTab.url, updatedURL = WebServer.sharedInstance.updateLocalURL(url) {
-                    let tab = self.addTab(request: NSURLRequest(URL: updatedURL), flushToDisk: false, zombie: true)
-                    tab.screenshot = savedTab.screenshot
-                    if savedTab.isSelected {
-                        tabToSelect = tab
-                    }
-                }
+                let tab = self.addTab(flushToDisk: false, zombie: true)
+                tab.screenshot = savedTab.screenshot
+                tab.currentPage = savedTab.currentPage
+                tab.urls = savedTab.urls
```

Comment:
And then these two lines will just become `tab.sessionData = savedTab.sessionData`.